components/service/sync-logins/src/main/java/mozilla/components/service/sync/logins/AsyncLoginsStorage.kt

```diff
@@ -309,8 +332,11 @@ interface AsyncLoginsStorage : AutoCloseable {
      * Run some [block] which operates over an unlocked instance of [AsyncLoginsStorage].
      * Database is locked once [block] is done.
      *
+     * This method is thread safe.
+     *
      * @throws [InvalidKeyException] if the provided [key] isn't valid.
      */
+    @Synchronized
     suspend fun <T> withUnlocked(
```

Comment:
Please correct me if I'm wrong here @grigoryk, I'm not super confident in concurrency.

IIUC, this should be a safe and simple solution because:
- None of our calls rely on state set by each other to finish, so no risk of deadlocks
- Our access here is relatively fast, so queuing calls + synchronization overhead shouldn't have any substantial perf impact

If I'm wrong on either of these (or just missing something important) please let me know and I'm happy to reevaluate.